### DISCUSSION TITLE
Weird MongoDb collection naming caused by NextJs Web App

### DISCUSSION THREAD

**Discussion by 2Tako2** [vercel/community#7410] [Help]
### Summary

I have a TurboRepo with NextJs web app using MongoDb for database and hosted on Vercel.

After deploying the NextJs web app, new collections were generated with weird naming, e.g. "accounts" got named "ys", "stations got named "ds".

Even though the local build and the deployed one are using the same database, it creates a new collection with a single letter naming (plus s, defaulted by MongoDb)

I have tried using a different MongDb database connection, different user role, and different database name. I have also tried setting "swcMinify" to false in Next config. As well as running it in both Prod and Review environment.

I am suspecting Vercel is the cause of this, but cannot find a solution.

My packages version
- next: 14.2.4
-@typegoose/typegoose: 11.3.0
- mongoose: 7.3.1
- node: 18.18.2

### Steps to Reproduce

I dont think it can be recreated, but...

Create NextJs in a Turbo Repo, connect MongoDb with NextJs Api serverless backend,
create a record that initiate a collection
check collection name

**Comment by pawlean** [comment]
Hi @2Tako2!

This does sound unusual.

Could you create a minimal reproducible example and deploy it to see if the problem persists? This can help isolate whether it’s an issue with Vercel, your code, or the interaction between the two.

Other steps to investigate:

1. Make sure your collection names are correctly defined in your schemas and models without any dynamic renaming.

2. Verify that the environment variables in Vercel match your local environment, particularly for the database URI and related settings.

3. Double-check if any minification or obfuscation is happening during the build process that could affect collection names.

4. Add logging to see which collections are being accessed or created, and deploy this version to Vercel to check the logs for any clues.

5. Are Vercel-specific configurations affecting your build? Such as settings in `vercel.json` or the Vercel Dashboard.

6. Verify that user roles and permissions are correctly set and not causing any unusual behaviors.

7. Check for any middleware or plugins that might be renaming your collections unintentionally.

8. Check for version discrepancies.

**Reply by 2Tako2** [reply]
First of all, thank you for the reply!

It is quite difficult for me to create a minimal reproducible example as the system is not a simple one.

However, could you please expand on step 3 about minification or obfuscation and step 5 about vercel.json? I have tried setting "scwMinify" to false in my next.config.js, but it didnt seem to do anything.

With the other steps (other than step 3 and step 5), I believe I have verified them and ruled them out to be the cause of this issue.

Thank you for your help!

**Reply by pawlean** [reply]
Could you share any `vercel.json` configuration that you have?

**Reply by pawlean** [reply]
You've already tried setting `swcMinify` to `false`, but double-check if there's any other minification or obfuscation happening during the build process.

**Reply by 2Tako2** [reply]
I believe I don't have any vercel.json configuration, I simply deployed my web app through the internet on Vercel by linking my GitHub repo.

Could you please suggest how I could check for any minification or obfuscation happening during the build process? Thank you

Potentially, would I be able to contact someone that could gain access to my vercel deployment and investigate?

[a comment by an automated account is not shown]
